vignettes/epipredict.Rmd

```diff
@@ -1,470 +1,565 @@
-Basically, `{parsnip}` unifies the function signature across statistical models.
-For example, `lm()` "likes" to work with formulas, but `glmnet::glmnet()` uses
-`x` and `y` for predictors and response. `{parsnip}` is agnostic. Both of these
-do "linear regression". Above we switched from `lm()` to `xgboost()` without
-any issue despite the fact that these functions couldn't be more different.
-
-```{r, eval = FALSE}
-lm(formula, data, subset, weights, na.action,
-  method = "qr",
-  model = TRUE, x = FALSE, y = FALSE, qr = TRUE, singular.ok = TRUE,
-  contrasts = NULL, offset, ...
-)
 
-xgboost(
-  data = NULL, label = NULL, missing = NA, weight = NULL,
-  params = list(), nrounds, verbose = 1, print_every_n = 1L,
-  early_stopping_rounds = NULL, maximize = NULL, save_period = NULL,
-  save_name = "xgboost.model", xgb_model = NULL, callbacks = list(),
-  ...
-)
-```
+## Fitting multi-key panel data
 
-`{epipredict}` provides a few engines/modules (the flatline forecaster and
-quantile regression), but you should be able to use any available models
-listed [here](https://www.tidymodels.org/find/parsnip/).
+If multiple keys are set in the `epi_df` as `other_keys`,
+`arx_forecaster` will automatically group by those in addition to the required geographic key.
+For example, predicting the number of graduates in each of the categories in `grad_employ_subset` from above:
 
-To estimate (fit) a preprocessed model, one calls `fit()` on the `epi_workflow`.
+```{r multi_key_forecast, warning=FALSE}
+# only fitting a subset, otherwise there are ~550 distinct pairs, which is bad for plotting
+edu_quals <- c("Undergraduate degree", "Professional degree")
+geo_values <- c("Quebec", "British Columbia")
 
-```{r}
-ewf <- epi_workflow(er, linear_reg()) %>% fit(jhu)
-```
+grad_employ <- grad_employ_subset |>
+    filter(time_value < 2017) |>
+    filter(edu_qual %in% edu_quals, geo_value %in% geo_values)
 
-### Postprocessing
+grad_employ
 
-To stretch the metaphor of preparing a cake to its natural limits, we have
-created postprocessing functionality called "frosting". Much like the recipe,
-each postprocessing operation is a "layer" and we "slather" these onto our
-baked cake. To fix ideas, below is the postprocessing `frosting` for
-`arx_forecaster()`
+grad_forecast <- arx_forecaster(
+  grad_employ |>
+    filter(time_value < 2017),
+  outcome = "num_graduates",
+  predictors = c("num_graduates"),
+  args_list = arx_args_list(
+    lags = list(c(0, 1, 2)),
+    ahead = 1
+  )
+)
+# and plotting
+autoplot(
+  grad_forecast$epi_workflow,
+  grad_forecast$predictions,
+  grad_employ,
+)
+```
 
-```{r}
-extract_frosting(out_q$epi_workflow)
+The 8 graphs represent all combinations of the `geo_values` (`"Quebec"` and `"British Columbia"`), `edu_quals` (`"Undergraduate degree"` and `"Professional degree"`), and age brackets (`"15 to 34 years"` and `"35 to 64 years"`).
+
+## Fitting a non-geo-pooled model
```

Comment:
Here, "Estimating models without geo-pooling" (technically, there is one model
per geo, so multiple models, unless you think of a larger model class that
encapsulates this flexibility)